tests/test_init.py

```diff
@@ -0,0 +1,69 @@
+"""
+Test cases for __init__.py.
+"""
+# Copyright (c) 2018 Red Hat, Inc. All rights reserved. This copyrighted
+# material is made available to anyone wishing to use, modify, copy, or
+# redistribute it subject to the terms and conditions of the GNU General Public
+# License v.2 or later.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+from xmlrpclib import ServerProxy
+import unittest
+
+import skt
+
+
+class TestInit(unittest.TestCase):
+    """Test cases for skt's __init__.py"""
+
+    def test_stringify_with_integer(self):
+        """Ensure stringify() can handle an integer."""
+        myinteger = int(42)
+        result = skt.stringify(myinteger)
+        self.assertIsInstance(result, str)
+        self.assertEqual(result, str(myinteger))
+
+    def test_stringify_with_string(self):
+        """Ensure stringify() can handle a plain string."""
+        mystring = "Test text"
+        result = skt.stringify(mystring)
+        self.assertIsInstance(result, str)
+        self.assertEqual(result, mystring)
+
+    def test_stringify_with_unicode(self):
+        """Ensure stringify() can handle a unicode byte string."""
+        myunicode = unicode("Test text")
+        result = skt.stringify(myunicode)
+        self.assertIsInstance(result, str)
+        self.assertEqual(result, myunicode.encode('utf-8'))
+
+    def test_parse_bad_patchwork_url(self):
+        """Ensure parse_patchwork_url() handles a parsing exception."""
+        patchwork_url = "garbage"
+        with self.assertRaises(Exception) as context:
+            skt.parse_patchwork_url(patchwork_url)
+
+            self.assertTrue(
+                "Can't parse patchwork url: '{}'".format(patchwork_url)
+                in context.exception
+            )
```

Comment:
Hmm, I'm not quite getting how this works. Should this really be inside `with`?

Reply:
That's the only spot were the `context` variable is available to me.

Reply:
Hmm, then what would happen if the assertion checking throws the same exception, and why does [this code in the documentation](https://docs.python.org/2/library/unittest.html#unittest.TestCase.assertRaises) has the test outside? Sorry, my context manager-fu is weak.

Reply:
Weird, it does work outside the `with:`. I removed the indention in the latest push.